### PR TITLE
[LI-HOTFIX] Add a Github Actions workflow to trigger build and run all tests upon a tag creation/push

### PR DESCRIPTION
**Pull request by Lincong**
Add a Github Actions workflow to trigger build and run all tests upon a tag creation/push

TICKET = N/A
LI_DESCRIPTION = LIKAFKA-34202
EXIT_CRITERIA = N/A

### CODE REVIEW

**Comment by Lincong** [comment]
Note that this change `does not` actually upload the archive since there are several things to verify before we enable the feature to upload the archive. Please see comments in the `TODO` section at the bottom of the added `yml` file. Thanks!

File: .github/workflows/build-and-upload-archives-upon-creating-tags.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          fetch-depth: 0
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Set up release version env variable
+        uses: actions/checkout@v2
+        run: |
+          echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+          echo "Release version: ${{ env.RELEASE_VERSION }}"
```
Comment (gitlw) [inline .github/workflows/build-and-upload-archives-upon-creating-tags.yml:49]: This [doc](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions) says
"The action that creates or updates the environment variable does not have access to the new value, but all subsequent actions in a job will have access."

File: .github/workflows/build-and-upload-archives-upon-creating-tags.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Build with Gradle and run all tests
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (gitlw) [inline .github/workflows/build-and-upload-archives-upon-creating-tags.yml]: Is this necessary? Which part requires the GITHUB_TOKEN?

Comment (Lincong) [reply .github/workflows/build-and-upload-archives-upon-creating-tags.yml]: It is not necessary.

File: .github/workflows/build-and-upload-archives-upon-creating-tags.yml (hunk 1 above)
Comment (Lincong) [reply .github/workflows/build-and-upload-archives-upon-creating-tags.yml:49]: Good catch!

**Review by gitlw — APPROVED** [review]
LGTM, thanks!

**Pull request merged**